coreff_ellipro/ellipro.py

```diff
@@ -0,0 +1,221 @@
+            self.company_status
+        )
+        ET.SubElement(search_options, "establishmentStatus").text = (
+            self.establishment_status
+        )
+
+
+@dataclass
+class Order:
+    company_id: str
+    product_id: str
+    product_version = "1"
+    output_method = "raw"
+    format = "XML"
+
+    def set_element(self, root):
+        request = ET.SubElement(root, "request")
+        ET.SubElement(request, "country", attrib={"code": "FRA"})
+        ET.SubElement(request, "id", attrib={"type": "src"}).text = (
+            self.company_id
+        )
+        ET.SubElement(
+            request,
+            "product",
+            attrib={"range": self.product_id, "version": self.product_version},
+        )
+        delivery_options = ET.SubElement(request, "deliveryOptions")
+        ET.SubElement(delivery_options, "outputMethod").text = (
+            self.output_method
+        )
+        ET.SubElement(delivery_options, "format").text = self.format
+
+
+@dataclass
+class Catalogue:
+    company_id: str
+    country_code = "FRA"
+
+    def set_element(self, root):
+        request = ET.SubElement(root, "request")
+        ET.SubElement(request, "country", attrib={"code": self.country_code})
+        ET.SubElement(request, "id", attrib={"type": "src"}).text = (
+            self.company_id
+        )
+
+
+# * send a request to Ellipro and returns an elementTree object
+def search(admin, request, request_type, lang="FR", version="2.2"):
+    headers = {"Content-Type": "application/xml"}
+    root = ET.Element(
+        f"{request_type}Request", attrib={"lang": lang, "version": version}
+    )
+    admin.set_element(root)
+    request.set_element(root)
+    body = ET.tostring(root)
+
+    request_result = requests.post(
+        f"https://services-test.data-access-gateway.com/1/rest/{request_type}",
+        data=body,
+        headers=headers,
+    )
+    return ET.fromstring(request_result.text)
+
+
+# * search for a lambda company, checks if the request send back an error, then returns True and the error
+def connection_check(admin):
+    request_type = RequestType.SEARCH
+    request = Search(
+        SearchType.NAME,
+        "Bonduelle",
+        "1",
+        IdType.ESTB,
+    )
+    response = search(admin, request, request_type.value)
+    response = ET.fromstring(response)
+
+    for search_response in response.iter("svcSearchResponse"):
+        error = {}
+        throw_error = (
+            search_response.findall("result")[0].attrib["code"] == "ERR"
+        )
+        if throw_error:
+            error["message"] = search_response.findall("result/minorMessage")[
+                0
+            ].text
+            error["infos"] = search_response.findall("result/additionalInfo")[
+                0
+            ].text
+    return throw_error, error
+
+
+# * create a list of suggestions based on a SvcSearch request response in a form of an elementTree object
```

Comment:
Description comment need to be in specific style: https://peps.python.org/pep-0257/